src/webview/columns/data-column.tsx

```diff
@@ -31,67 +55,240 @@ export class DataColumn implements ColumnContribution {
+        return <span className='single-word' key={word.address.toString(16)}>{
+            word.bits.map(b => this.renderEightBits(b))
+        }</span>;
+    }
+
+    protected renderEightBits(bit: DataColumnBit): React.ReactNode {
+        const { content, className, style, title } = this.getBitAttributes(bit);
         return <span
             style={style}
-            key={offset.toString(16)}
+            key={bit.offset.toString(16)}
             className={className}
-            data-id={offset}
+            data-id={bit.offset}
             title={title}
         >
             {content}
         </span>;
     }
 
-    protected getBitAttributes(memory: Memory, currentAdress: bigint, offset: number): FullNodeAttributes {
+    protected getBitAttributes(bit: DataColumnBit): FullNodeAttributes {
         return {
             className: 'eight-bits',
-            style: decorationService.getDecoration(currentAdress)?.style,
-            content: (memory.bytes[offset] ?? 0).toString(16).padStart(2, '0')
+            style: decorationService.getDecoration(bit.address)?.style,
+            content: bit.value
         };
     }
+
+    protected enableEdit: React.MouseEventHandler = event => {
+        this.setState(prev => ({ ...prev, isEdit: true }));
+        event.stopPropagation();
+    };
+
+    protected disableEdit = () => {
+        this.setState(prev => ({ ...prev, isEdit: false, value: this.asLine(this.props.group) }));
+    };
+
+    protected onBlur: React.FocusEventHandler<HTMLInputElement> = () => {
+        this.disableEdit();
+        this.submitChanges();
+    };
+
+    protected onKeyDown: React.KeyboardEventHandler<HTMLInputElement> = event => {
+        switch (event.key) {
+            case 'Escape': {
+                this.disableEdit();
+                break;
+            }
+            case 'Enter': {
+                this.submitChanges();
+            }
+        }
+        event.stopPropagation();
+    };
+
+    protected onChange: React.ChangeEventHandler<HTMLInputElement> = event => {
+        this.setState(prev => ({ ...prev, value: event.target.value }));
+    };
+
+    protected async submitChanges(): Promise<void> {
+        const newData = this.processData(this.state.value);
+        const original = this.processData(this.asLine(this.props.group));
+
+        if (newData && newData !== original) {
+            const converted = Buffer.from(newData, 'hex').toString('base64');
+            await this.props.writeMemory?.({
+                memoryReference: toHexStringWithRadixMarker(this.props.group.startAddress),
+                data: converted
+            });
+        }
+
+        this.disableEdit();
+    }
+
+    protected processData(data: string): string {
+        // Revert Endianness
+        if (this.props.options.endianness === Endianness.Little) {
+            const chunks = data.padStart(this.renderableCharacters, '0').match(/.{1,2}/g) || [];
```

Comment:
What's the significance of the `.` and why would 2 appear?

Reply:
I'm just splitting here the whole string into chunks of length 2, so that I can reverse each chunk. (The `1` is not necessary, will be removed)